pyproject.toml

```diff
@@ -329,3 +292,37 @@ branch = "fix_signal_issue"
 [tool.poetry.group.dev.dependencies]
```

Comment:
Seems like a legit solution. 
The brute force way to check if this would fix the build issue is to do a build re-run of the CI workflow on this PR, around 10-15 times and check if the install dependencies fails.
Also think if re-run should still check dependency install for python 3.7 since we are dropping support for it in another PR.

Reply:
@radovanZRasa Running CI 10-15 times will take forever. I've ran it three times successfully so far and haven't noticed issues. I think I'll take this forward and open another PR if any issues come up again